From eb29f2ac9a401146bb09bf682c6f2d2093de869f Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Sat, 22 Jan 2022 11:27:54 +0100
Subject: [PATCH 101/261] efi: Correct struct grub_efi_boot_services

The UEFI specification defines that the EFI_BOOT_SERVICES.Exit(() service may return
EFI_SUCCESS or EFI_INVALID_PARAMETER. So it cannot be __attribute__((noreturn)).

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 include/grub/efi/api.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/grub/efi/api.h b/include/grub/efi/api.h
index f1a52210c..6c8d06e15 100644
--- a/include/grub/efi/api.h
+++ b/include/grub/efi/api.h
@@ -1196,7 +1196,7 @@ struct grub_efi_boot_services
   (*exit) (grub_efi_handle_t image_handle,
 	   grub_efi_status_t exit_status,
 	   grub_efi_uintn_t exit_data_size,
-	   grub_efi_char16_t *exit_data) __attribute__((noreturn));
+	   grub_efi_char16_t *exit_data);
 
   grub_efi_status_t
   (*unload_image) (grub_efi_handle_t image_handle);
-- 
2.36.1

